syntax = "proto3";

package auth;

option java_multiple_files = true;
option java_package = "com.stdiscm.auth";

enum Role {
  ROLE_UNSPECIFIED = 0;
  STUDENT = 1;
  FACULTY = 2;
}

message User {
  string id = 1;
  string name = 2;
  string email = 3;
  Role role = 4;
}

message TokenValidationRequest {
  string token = 1;
}

message TokenValidationResponse {
  bool valid = 1;
  string user_id = 2;
  string email = 3;
  Role role = 4;
  string reason = 5;
}

message GetUserRequest {
  string user_id = 1;
}

message GetUserResponse {
  User user = 1;
}

service AuthService {
  rpc ValidateToken (TokenValidationRequest) returns (TokenValidationResponse);
  rpc GetUser (GetUserRequest) returns (GetUserResponse);
}
